src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -39,25 +39,29 @@ class DirectStepI : public StepImp<IOUAmount, IOUAmount, DirectStepI>
     AccountID src_;
     AccountID dst_;
     Currency currency_;
-    // Transfer fees are never charged when sending directly to or from an
-    // issuing account
-    bool noTransferFee_;
+
+    // Charge transfer fees whan the prev step redeems
+    Step const* const prevStep_ = nullptr;
+
     beast::Journal j_;
 
     struct Cache
     {
         IOUAmount in;
         IOUAmount srcToDst;
         IOUAmount out;
+        bool srcRedeems;
 
         Cache () = default;
```

Comment:
Do you really need a default constructor?  If I change this line to `Cache () = delete;` everything builds fine.  The prospect of a bunch of uninitialized values makes me nervous.